warehouse/accounts/views.py

```diff
@@ -814,3 +818,52 @@ def edit_profile_button(user, request):
 )
 def profile_public_email(user, request):
     return {"user": user}
+
+
+@view_config(
+    require_methods=["POST"],
+    require_csrf=False,
+    renderer="json",
+    route_name="accounts.github-disclose-token",
+    headers=["GITHUB-PUBLIC-KEY-IDENTIFIER", "GITHUB-PUBLIC-KEY-SIGNATURE"],
```

Comment:
This almost feels like overkill for something that would only be an extra two lines in the view body:

```
if key_id is None or signature is None:
    return Response(status=400)
```

If we really think we want to keep this for future use, I'd recommend naming it something like `require_headers` instead, and adding some additional comments/documentation to the predicate. For example, it's not clear to me what a request with incorrect headers would get in response.

Reply:
I had (what I understood as) the opposite suggestion earlier in this review...

I’ll do my best to make everyone happy but if you can make sure that choice will be final, I’d hate to redo the same code a 4th time :)
For what it’s worth, it’s kinda what I had at first.

Reply:
Are you referring to https://github.com/pypa/warehouse/pull/7124#discussion_r414072228? I think that discussion was more about whether it was possible, not whether it was advisable 🙂 

Since you've already done the work though, I think it's fine to keep it unless the other maintainers (@dstufft / @ewdurbin) object for some reason. I do think we should be more clear about it's purpose though, just `headers` is not descriptive enough.

Reply:
> For example, it's not clear to me what a request with incorrect headers would get in response.

As far as I can tell, a view without those headers will not be considered during URL matching, and thus I'd expect 404.